### ISSUE TITLE
Sharding is posting 1 shard server count

### ISSUE THREAD

**Issue by pizzafox** [T0TProduction/BLAPI#11]
My bot is on thousands of servers and has 4 shards. BLAPI is sending to server listings that support shards and don't that it has 4 shards and is only on 912 servers (the first shard's server count).

I am running the latest version of BLAPI from GitHub and I'm not using manual posting.

Why is this happening and how can I fix this?

**Comment by MattIPv4** [comment]
@pizzafox Could you try using postman to send your stats post directly to the BotBlock endpoint?
Would like to ensure this isn't an issue I'm causing.

**Comment by T0TProduction** [comment]
Iirc i havent added shard posting to manual post yet, so that won't help
I'll check if im doing something wrong in a few days, when im back

**Comment by MattIPv4** [comment]
@T0TProduction I meant using postman, not your lib. To ensure it isn't an issue with the BotBlock API.

**Comment by pizzafox** [comment]
Ran via Postman. Results:
```json
{
  "success": {
    "discordbot.world": [
      200,
      "{\"success\":true,\"message\":\"Stats updated\"}",
      "{\"shards\":[911,909,897,849]}"
    ],
    "botsfordiscord.com": [
      0,
      false,
      "{\"server_count\":3566}"
    ],
    "bots.ondiscord.xyz": [
      204,
      "",
      "{\"guildCount\":3566}"
    ],
    "discordbots.org": [
      200,
      "{}",
      "{\"shard_count\":4,\"shards\":[911,909,897,849]}"
    ],
    "discord.services": [
      200,
      "{\"success\": true}",
      "{\"server_count\":3566}"
    ],
    "bots.discord.pw": [
      204,
      "",
      "{\"server_count\":3566,\"shard_count\":4}"
    ],
    "botlist.space": [
      200,
      "{\n    \"code\": 200,\n    \"message\": \"Successful at posting server count\"\n}",
      "{\"shards\":[911,909,897,849]}"
    ]
  },
  "failure": []
}
```

**Comment by T0TProduction** [comment]
@pizzafox you posted all servercounts within a single post?
Ofc blapi cant do that, you only have one servercount per code running
So sites that "support sharding" but need you to post every single count inside a combined post, obv won't work

(I also strongly disagree that those sites support sharded botcount posting, because they literally don't, they just show the data if you yourself aggregated it 🤔)

**Comment by pizzafox** [comment]
What do you mean by "you only have one servercount per code running"?

**Comment by T0TProduction** [comment]
Blapi is used in the code a single shard is running on, isnt it🤔

**Comment by pizzafox** [comment]
My first thought was that this was the case, but it isn't. The code is running on all shards.

**Comment by T0TProduction** [comment]
Wait what🤔
How is it running on all shards
Isnt the whole sharding to split the load between smaller instances?

**Comment by pizzafox** [comment]
Yes it is. The BLAPI.handle function is getting run on every shard. Each shard handles around 900 servers. Is this the proper implentation for BLAPI with sharding?

**Comment by T0TProduction** [comment]
Yes it is. BLAPI then posts that Count, for every single shard, and lists that *really* Support sharding then combine those postst
lists that need you to post an Array of guildcounts, Like you did in postman, won't work that way, but BLAPI won't know that array in this setup

**Comment by T0TProduction** [comment]
So, in theory, the wrong guildcount should only be on a few select lists, while others display it correctly.
I'll make a list of those when im back if you want me to

Ofc there could still be something wrong on BLAPIs side

**Comment by pizzafox** [comment]
When BLAPI handles posting the server counts with shards to any listing, it shows 4 shards with the server count being ${the first shard's server count}. I just want my server count to be accurate.

**Comment by T0TProduction** [comment]
OK that should **not** be the case for *every*(or any) site
I'll double check the code after reviewing all the issues when im back (soon™️)

**Comment by T0TProduction** [comment]
@pizzafox could you try posting different stats than your real ones (to see what changed and worked) via botBlock and postman, but in the form BLAPI would use? (4 posts with each only their own shard data, not the aggregated)
It looks to me that BLAPI forwards the correct data to botBlock

**Comment by MattIPv4** [comment]
@T0TProduction The postman request made appears to have the following data in the request
server_count: 3566
shards: [911, 909, 897, 849]
shard_count: 4

It appears it was a single request as would be expected for bot lists that accept a shards array.

**Comment by pizzafox** [comment]
This is resolved with PR #14/#15. Waiting for minor bug fixes on it and then merge with master.

**Comment by T0TProduction** [comment]
merged #14
